src/txn/txn_timestamp.c

```diff
@@ -403,6 +403,18 @@ __wt_txn_global_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
         return (0);
 
 set:
+    /*
+     * Even if the timestamps have been forcibly set, they must always satisfy the condition that
+     * oldest <= stable. Don't fail as MongoDB violates this rule in very specific scenarios.
+     */
+    if ((has_oldest || has_stable) && (has_oldest || txn_global->has_oldest_timestamp) &&
+      (has_stable || txn_global->has_stable_timestamp) && oldest_ts > stable_ts) {
```

Comment:
I don't think this is correct. With force, we bypass the code that sets oldest_ts and stable_ts.

Reply:
```
    /*
     * First do error checking on the timestamp values. The oldest timestamp must always be less
     * than or equal to the stable timestamp. If we're only setting one then compare against the
     * system timestamp. If we're setting both then compare the passed in values.
     */
    if (!has_oldest && txn_global->has_oldest_timestamp)
        oldest_ts = last_oldest_ts;
    if (!has_stable && txn_global->has_stable_timestamp)
        stable_ts = last_stable_ts;
```

Reply:
You are right, this is wrong.

Reply:
Fixed in 5ab5c5bee3865c89ba00dd8232ebcb92405d033f.